bin/migrate_to_single_table.py

```diff
@@ -0,0 +1,319 @@
+        Session = sessionmaker(bind=engine)
+
+        self.session = Session()
+
+    def index_record_to_new_table(self, batch_size=1000):
+        """
+        Collect records from index_record table, collect additional info from multiple tables and bulk insert to new record table.
+        """
+        try:
+            total_records = self.session.query(IndexRecord).count()
+            last_seen_guid = None
+            count = 0
+
+            while True:
+                if last_seen_guid is None:
+                    records = (
+                        self.session.query(IndexRecord)
+                        .order_by(IndexRecord.did)
+                        .limit(batch_size)
+                        .all()
+                    )
+                else:
+                    records = (
+                        self.session.query(IndexRecord)
+                        .order_by(IndexRecord.did)
+                        .filter(IndexRecord.did > last_seen_guid)
+                        .limit(batch_size)
+                        .all()
+                    )
+
+                if not records:
+                    break
+
+                records_to_insert = []
+
+                for record in records:
+                    hashes = self.get_index_record_hash(record.did)
+                    urls = self.get_urls_record(record.did)
+                    url_metadata = self.get_urls_metadata(record.did)
+                    acl = self.get_index_record_ace(record.did)
+                    authz = self.get_index_record_authz(record.did)
+                    alias = self.get_index_record_alias(record.did)
+                    metadata = self.get_index_record_metadata(record.did)
+                    records_to_insert.append(
+                        Record(
+                            guid=record.did,
+                            baseid=record.baseid,
+                            rev=record.rev,
+                            form=record.form,
+                            size=record.size,
+                            created_date=record.created_date,
+                            updated_date=record.updated_date,
+                            content_created_date=record.content_created_date,
+                            content_updated_date=record.content_updated_date,
+                            file_name=record.file_name,
+                            version=record.version,
+                            uploader=record.uploader,
+                            hashes=hashes,
+                            urls=urls,
+                            url_metadata=url_metadata,
+                            acl=acl,
+                            authz=authz,
+                            alias=alias,
+                            record_metadata=metadata,
+                        )
+                    )
+                while records_to_insert:
+                    try:
+                        self.session.bulk_save_objects(records_to_insert)
```

Comment:
SQLAlchemy lists this as legacy https://docs.sqlalchemy.org/en/20/orm/session_api.html#sqlalchemy.orm.Session.bulk_save_objects for 2.0 and beyond. I think since this is a one time run and we're not ported to 2 yet, we're okay